dns-compatible-image-names.md

```diff
@@ -0,0 +1,30 @@
+# DNS-compatible image names
+
+This is version 1 of this specification.
+
+The [X.509 Public Key Infrastructure][X.509] provides a well-established mechanism for trusted namespacing of [domain names][rfc5890].
+Protocols interested in leveraging that infrastructure need to be able to extract domain names from image names.
+This specification provides one approach for that extraction.
+
+This specification defines image names compatible with DNS using the following [ABNF][]:
+
+```ABNF
+dns-compatible-image-name = host "/" path-rootless [ "#" fragment ]
+```
```

Comment:
This conforms to https://tools.ietf.org/html/rfc3986#section-3, that seems to be compatible to both DNS and non-DNS case. The scenario without DNS is not mentioned in your PR, but I think it can also be supported by our discussing protocol.

Reply:
> This conforms to https://tools.ietf.org/html/rfc3986#section-3, that seems to be compatible to both DNS and non-DNS case.

I agree that these names could be resolved outside of DNS (e.g. via an `/etc/hosts` file).  I've pushed 8d7eb73a → 292435dd squashing some fixups that change `dns-compatible-image-name` to `host-based-image-name`, mention IP address forms, and SHOULD DNS-comatible `host` values (because otherwise using X.509 is hard).